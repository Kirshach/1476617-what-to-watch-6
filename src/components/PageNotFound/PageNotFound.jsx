import React from 'react';

import PageFooter from '../PageFooter/PageFooter';
import PageHeader from '../PageHeader/PageHeader';

const PageNotFound = () => {
  return (
    <div className="user-page">
      <PageHeader className="user-page__head" noUserBlock/>

      <main className="info-screen">
        <h2 className="info-screen__heading">404</h2>
        <p className="info-screen__message">Sorry, the page you&apos;re looking for could not be found</p>
        <footer className="info-screen__footer">It could be you, it could be us...</footer>
      </main>

      <PageFooter />
    </div>
  );
};

PageNotFound.propTypes = {};

export default PageNotFound;
