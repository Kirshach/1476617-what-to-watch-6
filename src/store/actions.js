export const ActionType = {
  SET_GENRE: `films/setGenre`,
};

export const setGenre = (payload) => ({
  type: ActionType.SET_GENRE,
  payload
});
