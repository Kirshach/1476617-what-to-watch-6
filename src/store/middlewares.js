import {history} from '../history';

export const redirectActionType = `app/redirect`;

export const redirectAction = (payload) => ({
  type: redirectActionType,
  payload,
});

export const redirectMiddleware = (_store) => (next) => (action) => {
  if (action.type === redirectActionType) {
    history.push(action.payload);
  }
  return next(action);
};
