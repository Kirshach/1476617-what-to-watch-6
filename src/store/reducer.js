import {ActionType} from './actions';
import {Genres} from '../const';
import {films} from '../mocks/films';


const initialState = {
  films,
  genre: Genres.allGenres,
  genres: Object.values(Genres),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };
    default:
      return state;
  }
};

export {reducer};
