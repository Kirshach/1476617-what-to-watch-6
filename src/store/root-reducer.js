import {combineReducers} from 'redux';

import {appReducer} from './app/reducer';
import {domainReducer} from './domain/reducer';

export const DOMAIN = `domain`;
export const APP = `app`;

const rootReducer = combineReducers({
  [DOMAIN]: domainReducer,
  [APP]: appReducer,
});

export {rootReducer};

export const RootNS = {DOMAIN, APP};
