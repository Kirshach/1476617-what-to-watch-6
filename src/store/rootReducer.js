import {combineReducers} from 'redux';

import {appReducer} from './app/reducer';
import {domainReducer} from './domain/reducer';

const DOMAIN = `domain`;
const APP = `app`;

const rootReducer = combineReducers({
  [DOMAIN]: domainReducer,
  [APP]: appReducer,
});

export {rootReducer};

export const RootNS = {DOMAIN, APP};
