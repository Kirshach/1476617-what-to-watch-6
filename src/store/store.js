import {createStore, applyMiddleware} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import {redirectMiddleware, handleAPIErrorMiddleware} from './middlewares';
import {checkAuthThunk} from '../store/app/auth/thunks';
import {fetchFilmsThunk} from '../store/domain/thunks';
import {rootReducer as reducer} from '../store/rootReducer';

import {api} from '../api/api';

const store = createStore(
    reducer,
    composeWithDevTools(
        applyMiddleware(
            handleAPIErrorMiddleware,
            redirectMiddleware,
            thunk.withExtraArgument(api),
        ),
    ),
);

store.dispatch(checkAuthThunk());
store.dispatch(fetchFilmsThunk());

export {store};
