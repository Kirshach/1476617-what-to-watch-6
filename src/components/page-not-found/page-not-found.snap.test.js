import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {useSelector} from 'react-redux';
import {render} from '@testing-library/react';

import {store} from '../../mocks/store/store';

import PageNotFound from './page-not-found';

jest.mock(`react-redux`);

test(`<PageNotFound /> renders correctly`, () => {
  useSelector.mockImplementation((selector) => selector(store));
  const {container} = render(
      <MemoryRouter>
        <PageNotFound />
      </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
