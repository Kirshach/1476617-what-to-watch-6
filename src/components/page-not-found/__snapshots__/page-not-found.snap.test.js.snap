// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> renders correctly 1`] = `
<div>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
    </header>
    <main
      class="info-screen"
    >
      <h2
        class="info-screen__heading"
      >
        404
      </h2>
      <p
        class="info-screen__message"
      >
        Sorry, the page you're looking for could not be found
      </p>
      <footer
        class="info-screen__footer"
      >
        It could be you, it could be us...
      </footer>
    </main>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
